scripts/llnl/build_src.py

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+"exec" "python" "-u" "-B" "$0" "$@"
+
+# Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+# other Serac Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+
+"""
+ file: build_src.py
+
+ description: 
+  Builds Serac with the host-configs for the current machine.
+
+"""
+
+from common_build_functions import *
+
+from optparse import OptionParser
+
+
+def parse_args():
+    "Parses args from command line"
+    parser = OptionParser()
+    # Location of source directory to build
+    parser.add_option("-d", "--directory",
+                      dest="directory",
+                      default="",
+                      help="Directory of source to be built (Defaults to current)")
+    # Whether to build a specific hostconfig
+    parser.add_option("--host-config",
+                      dest="hostconfig",
+                      default="",
+                      help="Build a specific hostconfig (defaults to env $HOST_CONFIG, then to $SYS_TYPE/$COMPILER available)")
+
+    ###############
+    # parse args
+    ###############
+    opts, extras = parser.parse_args()
+    # we want a dict b/c the values could 
+    # be passed without using optparse
+    opts = vars(opts)
+    return opts
+
+
+def main():
+    opts = parse_args()
+
+    # Determine source directory to be built
+    if os.environ.get("UBERENV_PREFIX") != None:
+        repo_dir = os.environ["UBERENV_PREFIX"]
+        if not os.path.isdir(repo_dir):
+            print("[ERROR: Given environment variable 'UBERENV_PREFIX' is not a valid directory]")
+            print("[    'UBERENV_PREFIX' = %s]" % repo_dir)
+            return 1
+    if opts["directory"] != "":
+        repo_dir = opts["directory"]
+        if not os.path.isdir(repo_dir):
+            print("[ERROR: Given command line variable '--directory' is not a valid directory]")
+            print("[    '--directory' = %s]" % repo_dir)
+            return 1
+    else:
+        repo_dir = get_repo_dir()
+
+    try:
+        original_wd = os.getcwd()
+        os.chdir(repo_dir)
+        timestamp = get_timestamp()
+        # Command-line arg has highest priority
+        if opts["hostconfig"] != "":
+          hostconfig = opts["hostconfig"]
+        
+        # Then check the environment
+        elif os.environ.get("HOST_CONFIG") != None:
```

Comment:
my distaste for magic environment variables is high...

Reply:
Don’t blame him, he ported a "feature" of my script... :)

Reply:
i know.. but this is my script its infecting =) it's my baby...

Reply:
Get a vaccine...
In bash it was a convenience, to avoid dealing with multiple arguments, and variables needed were exposed at the beginning of the script. That’s not a very good defense, but that is to say I won’t fight to keep the magic :)